src/sidebar/services/analytics.ts

```diff
@@ -0,0 +1,20 @@
+import type { APIService } from './api';
+
+/**
+ * @inject
+ */
+export class AnalyticsService {
+  private _api: APIService;
+
+  constructor(api: APIService) {
+    this._api = api;
+  }
+
+  trackApplyPendingUpdatesEvent(): void {
+    this._api.analytics.events
+      .create({}, { event: 'APPLY_PENDING_UPDATES' })
+      .catch(e =>
+        console.error('Could not track applying pending updates.', e),
```

Comment:
I would log a warning rather than an error, since nothing is broken for users. Actually recording the details here is probably not needed, since developers have that info in the browser console.

Reply:
> I would log a warning rather than an error, since nothing is broken for users.
👍🏼 

> Actually recording the details here is probably not needed, since developers have that info in the browser console.

Since the promise error is captured, there's actually no info logged automatically, so I would still log something.